packages/iov-keycontrol/src/userprofile.ts

```diff
@@ -192,27 +191,23 @@ export class UserProfile {
     return keyring.getAllIdentities();
   }
 
+  /**
+   * Signs a transaction using the profile's primary keyring. The transaction's
+   * creator field specifies the keypair to be used for signing.
+   */
   public async signTransaction(
     id: WalletId,
-    identity: PublicIdentity,
     transaction: UnsignedTransaction,
     codec: TxCodec,
     nonce: Nonce,
   ): Promise<SignedTransaction> {
-    // identity and the pair (transaction.chainId, transaction.signer) are redundant
-    // but we keep both in the interface to be consistent with appendSignature() where
-    // the original transaction creator is not the signer
```

Comment:
Why exactly is this no longer valid?

Reply:
We decided that it is not important to keep the signature of signTransaction() and appendSignature() the same. In the simple cases (signTransaction), the transaction creator is always the one who signs the transaction. This saves us an additional, redundant argument.